classes/components/forms/context/PKPDateTimeForm.inc.php

```diff
@@ -0,0 +1,161 @@
+			->addField(new FieldRadioInput('dateFormatLong', [
+				'label' => __('manager.setup.dateTime.longDate'),
+				'isRequired' => true,
+				'isMultilingual' => true,
+				'options' => $this->_setDateOptions($currentDateTime, [
+					'%B %e, %Y',
+					'%B %e %Y',
+					'%e %B %Y',
+					'%Y %B %e',
+				]),
+				'value' => $context->getDateTimeFormats('dateFormatLong'),
+				'groupId' => 'descriptions',
+			]))
+			->addField(new FieldRadioInput('timeFormat', [
+				'label' => __('manager.setup.dateTime.time'),
+				'isRequired' => true,
+				'isMultilingual' => true,
+				'options' => $this->_setDateOptions($currentDateTime, [
+					'%I:%M %p',
+					'%I:%M %P',
+					'%k:%M',
+				]),
+				'value' => $context->getDateTimeFormats('timeFormat'),
+				'groupId' => 'descriptions',
+			]))
+			->addField(new FieldRadioInput('datetimeFormatShort', [
+				'label' => __('manager.setup.dateTime.shortDateTime'),
+				'isRequired' => true,
+				'isMultilingual' => true,
+				'options' => array_map(function ($value) use ($context, $currentDateTime, $localizedOptions) {
+					$locale = array_search($value, $localizedOptions);
+					setlocale(LC_TIME, $locale . '.utf8');
+					$optionValue = $context->getLocalizedDateFormatShort($locale) . ' ' . $context->getLocalizedTimeFormat($locale);
+					return [
+						[
+							'value' => $optionValue,
+							'label' => strftime($optionValue, $currentDateTime),
+						],
+						[
+							'isInput' => true,
+							'label' => __('manager.setup.dateTime.custom'),
+						]
+					];
+				}, $localizedOptions),
+				'value' => $context->getDateTimeFormats('datetimeFormatShort'),
+				'groupId' => 'descriptions',
+			]))
+			->addField(new FieldRadioInput('datetimeFormatLong', [
+				'label' => __('manager.setup.dateTime.longDateTime'),
+				'isRequired' => true,
+				'isMultilingual' => true,
+				'options' => array_map(function ($value) use ($context, $currentDateTime, $localizedOptions) {
+					$locale = array_search($value, $localizedOptions);
+					setlocale(LC_TIME, $locale . '.utf8');
+					$optionValue = $context->getLocalizedDateFormatLong($locale) . ' - ' . $context->getLocalizedTimeFormat($locale);
+					return [
+						[
+							'value' => $optionValue,
+							'label' => strftime($optionValue, $currentDateTime),
+						],
+						[
+							'isInput' => true,
+							'label' => __('manager.setup.dateTime.custom'),
+						]
+					];
+				}, $localizedOptions),
+				'value' => $context->getDateTimeFormats('datetimeFormatLong'),
+				'groupId' => 'descriptions',
+			]));
+	}
+
+	/**
+	 * @param $locales array of supported locales
+	 * @param $currentDateTime string current date and time to show for demonstration
+	 * @param $optionValues array options to pass to the field
+	 * @return array
+	 */
+	private function _setDateOptions($currentDateTime, $optionValues)
+	{
+		$options = [];
+		foreach ($this->locales as $localeValue) {
+			$locale = $localeValue['key'];
+			setlocale(LC_TIME, $locale . '.utf8');
```

Comment:
I'm not familiar with `setLocale()`. Do we need to set the locale back to what it was before the form was generated?

Reply:
Have you tried using the third parameter of the `__()` function to translate into a particular locale? https://github.com/pkp/pkp-lib/blob/master/classes/i18n/PKPLocale.inc.php#L775

Reply:
Yes, I think you are right about setLocale, at least if follow documentation. Another problem is that in this form it works only on Linux. E.g., Windows accepts locale strings in a different format. An option could be `IntlDateFormatter` but I'm not an expert in this.

Reply:
Hmm, I don't think `__()` is suitable for this. It seems only iterate locale files and return value if found. The locale is set earlier in `PKPLocale::initialize`

Reply:
Alec will not better about the locale stuff. @asmecher do you have any idea how to get a locale key in a locale other than the current one?

If not we may want to figure out a new approach for the field -- maybe dropping the "custom" label altogether.

Reply:
You can use `__(..., $locale)`, but first you'll have to load the other locale. To do that, use:
```
AppLocale::requireComponents(COMPONENT_NAME_HERE, `xx_YY`);
```
The last parameter, when a string is specified, is the locale for which the component should be loaded; it defaults to the current locale and so is generally not specified.

(This weird flexible parameter type stuff was introduced to avoid endlessly typing `array(...)`, and is obviated by using the `[...]` syntax. I'd like to deprecate the wacky parameter processing in favour of a standard parameter list.)

Reply:
Here I'm using `setLocale()` to localize label with `strfrime`: https://github.com/Vitaliy-1/pkp-lib/blob/59900191f08c70428c31b74b3ff7ea8ec907e8b2/classes/components/forms/context/PKPDateTimeForm.inc.php#L150 I can't use `__()` for it. @asmecher, is it OK to use this approach with iteration through all locales and using `setLocale`? From the description, I see that it's maintained per process, does this have any potential side-effects (per Nate's comment I've added [a commit](https://github.com/Vitaliy-1/pkp-lib/commit/59900191f08c70428c31b74b3ff7ea8ec907e8b2) to return locale to the initial one)?